### ISSUE TITLE
Backward LSTM results mismatch between TF and ONNXRuntime

### ISSUE THREAD

**Issue by buddhapuneeth** [onnx/keras-onnx#523]
Backward LSTM results mismatch between TF and ONNXRuntime.
in TF runtime (tf.keras), say 1,2,3 are time frames.. output of reverse LSTM will be op3 op2 op1. where as in ONNXRuntime it will be op1 op2 op3. 

I think keras-onnx need to add a reverseV2 operator after backward LSTM in order to get matched results.
i.e. at
https://github.com/onnx/keras-onnx/blob/master/keras2onnx/ke2onnx/lstm.py#L213
After transpose we need to use reverseV2 or reverse sequence to match the order.

To reproduce:
[simple_lstm.txt](https://github.com/onnx/keras-onnx/files/4761010/simple_lstm.txt)
-rename the above file to .h5
-run with some random input on tf.keras backend and on onnxruntime (after conversion) and compare results.
-outputs are just reverse to each other.

Alternatively, you can try this:
code for reverse LSTM:
```
max_features = 10
maxlen = 10
model = Sequential()
model.add(Embedding(max_features, 128, input_length=maxlen))

model.add(LSTM(64, go_backwards=True, return_sequences = True))
model.add(Dense(1, activation='sigmoid'))
model.compile('adam', 'binary_crossentropy', metrics=['accuracy'])


from keras.models import load_model
model.save('simple_rev_lstm.h5')
```

To compare results with other inference engines and ONNXRuntime

```
# pip install onnxruntime
#env TF_KERAS=1
import numpy as np
import onnxruntime
from tensorflow.keras.models import load_model as load_model_tf_keras
np.random.seed(0)
input_data = np.random.randint(10, size=(2, 10))
# with CustomObjectScope({'GlorotUniform': glorot_uniform()}):
sess = onnxruntime.InferenceSession("simple_rev_lstm.onnx")
result = sess.run(["dense_1"], {'embedding_1_input': input_data.astype(np.float32)})
print("ONNX Runtime")
print(np.asarray(result[0]))

model = load_model_tf_keras('simple_rev_lstm.h5')
result = model.predict(input_data)
print("TF Runtime")
print(result)
```
OUTPUT:
```
ONNX Runtime
[[[0.5028585 ]
  [0.51010513]
  [0.5096171 ]
  [0.50567615]
  [0.50046337]
  [0.50202006]
  [0.5010086 ]
  [0.49436894]
  [0.4989641 ]
  [0.50032514]]

 [[0.50235206]
  [0.5045772 ]
  [0.504677  ]
  [0.5032916 ]
  [0.5016299 ]
  [0.50043035]
  [0.4991624 ]
  [0.50064933]
  [0.5028503 ]
  [0.5012636 ]]]

TF Runtime
[[[0.50032514]
  [0.4989641 ]
  [0.49436894]
  [0.5010086 ]
  [0.50202006]
  [0.50046337]
  [0.50567615]
  [0.5096171 ]
  [0.51010513]
  [0.5028585 ]]

 [[0.5012636 ]
  [0.5028503 ]
  [0.50064933]
  [0.4991624 ]
  [0.50043035]
  [0.5016299 ]
  [0.5032916 ]
  [0.504677  ]
  [0.5045773 ]
  [0.50235206]]]
```
Observation:
The values are in reverse to each other on axis 1.
Not sure which is correct behavior.

**Comment by wenbingl** [comment]
I read through LSTM spec: https://github.com/onnx/onnx/blob/master/docs/Operators.md#lstm and really didn't get any clue from the spec whether backward output should be reversed, so I don't know whether ORT result is correct or not.

Can you post an issue in ONNX issue list to get a clarification whether backward output should be reversed? Then we can decide how to follow up on this question.

**Comment by buddhapuneeth** [comment]
@wenbingl yes I already posted this question in ONNX https://github.com/microsoft/onnxruntime/issues/4187, waiting for the response.

**Comment by jiafatom** [comment]
In the onnxruntime post, you mentioned that you hack keras2onnx converter, what is the hack? If we want to support backward in converter side, seems that we need add onnx `ReverseSequence` op after lstm, and it looks a nontrivial change.

**Comment by buddhapuneeth** [comment]
@jiafatom what you mentioned is correct. I added ReverseSequence operator after LSTM conversion in keras2onnx if direction is 'reverse'. As you said, it is a nontrivial change. So looking for some confirmation from your side or ONNXRuntime side. If you agree with me, I can publish a PR of the change.

**Comment by jiafatom** [comment]
@buddhapuneeth in previous posts, we recommend you to post the issue in [onnx repo](https://github.com/onnx/onnx), not onnxruntime repo. In onnx spec, there is no mention the output should go forward or backward. So we want them to give a clarification. Thanks.

**Comment by wenbingl** [comment]
@buddhapuneeth , if this issue blocks you and we can't get onnx response quickly.
I can accept the idea that go_backward by default is False and it's the default behavior of ONNX (Suppose ONNX should adopt this idea as well). And the go_backward == True needs a reverse op. Do you have a PR for that?

**Comment by buddhapuneeth** [comment]
@jiafatom sure I will raise in ONNX repo as well.
@wenbingl I have code ready for this change. I can publish PR in a day or two.
